### ISSUE TITLE
Can't set long node name! using kubernetes

### ISSUE THREAD

**Issue by venkateshcontus** [processone/ejabberd#4184]
Hi,

    we started the ejabberd node with the below command

    sbin/ejabberdctl --node ejabberd@10.0.1.18 start it works and verified by logs and PSI+  . but we trying to check the sbin/ejabberdctl --node ejabberd@10.0.1.18 status getting error. Ip is created in kubernetes so we can't able to set in ejabberd.cfg file. 

## Environment

- ejabberd version: 24.02
- Erlang version: `24`
- OS: Linux (Debian)
- Installed from: source 

## Configuration (only if needed): grep -Ev '^$|^\s*#' ejabberd.yml

```yaml
loglevel: 4
...
```

## Errors from error.log/crash.log

--
=INFO REPORT==== 21-Mar-2024::10:05:33.064535 ===
Can't set long node name!
Please check your configuration

=SUPERVISOR REPORT==== 21-Mar-2024::10:05:33.064570 ===
    supervisor: {local,net_sup}
    errorContext: start_error
    reason: {'EXIT',nodistribution}
    offender: [{pid,undefined},
               {id,net_kernel},
               {mfargs,{net_kernel,start_link,
                                   [#{clean_halt => true,name => undefined,
                                      name_domain => longnames,
                                      supervisor => net_sup}]}},
               {restart_type,permanent},
               {significant,false},
               {shutdown,2000},
               {child_type,worker}]

{"Kernel pid terminated",application_controller,"{application_start_failure,kernel,{{shutdown,{failed_to_start_child,net_sup,{shutdown,{failed_to_start_child,net_kernel,{'EXIT',nodistribution}}}}},{kernel,start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,kernel,{{shutdown,{failed_to_start_child,net_sup,{shutdown,{failed_to_start_child,net_kernel,{'EXIT',nodistribution}}}}},{kernel,start,[normal,[]]}}})

Crash dump is being written to: /opt/ejabberd/var/log/ejabberd/erl_crash_20240321-100532.dump...done

## Bug description

Please, give us a precise description (what does not work, what is expected, etc.)

**Comment by prefiks** [comment]
Hello,

Would you be able to edit ejabberdctl script, and see if doing this change:
```
     # Erlang/OTP 23 or higher: use native dynamic node code
     # https://www.erlang.org/patches/otp-23.0#OTP-13812
-    echo undefined
+    if [ "$ERLANG_NODE" != "${ERLANG_NODE%.*}" ]; then
+      echo "undefined@${ERLANG_NODE#*@}"
+    else
+      echo "undefined"
+    fi
```
helps with your problem?

**Comment by Neustradamus** [comment]
@venkateshcontus: Have you seen the @prefiks comment?

**Comment by venkateshcontus** [comment]
> @venkateshcontus: Have you seen the @prefiks comment?

Yes, but we can't able to change it on runtime. We are using Kubernetes cluster architecture. So we can't do this change in the ejabberdctl script .

**Comment by prefiks** [comment]
This change was commited as 2c5a8f0860d874e71f7588842d3844b1b76db663. Since this seems to be fixed this problem in my tests, i am closing it.

**Comment by venkateshcontus** [comment]
thanks it's works me too
